miqa/core/rest/session.py

```diff
@@ -34,7 +74,9 @@ class Meta:
 
 
 class SessionViewSet(ReadOnlyModelViewSet):
-    queryset = Session.objects.all()
+    queryset = Session.objects.prefetch_related(
+        'experiments__scans__images', 'experiments__scans__notes'
+    )
```

Comment:
One design change we should make is to only use this queryset for the `retrieve` action rather than the `list` action. In practice, we'll only want to get this detailed result set for a single session at a time, rather than basically dumping all the data in the system.

We can use `get_queryset` method rather than the `queryset` property to dynamically set the queryset based on the request action.

Reply:
I think we would need to change the serializer class as well, because otherwise it will still go and gather foreign key relations. Does something like this look good?
```
    def get_queryset(self):
        if self.action == 'retrieve':
            return Session.objects.prefetch_related(
                'experiments__scans__images', 'experiments__scans__notes'
            )
        else:
            return Session.objects.all()

    def get_serializer_class(self):
        if self.action == 'retrieve':
            return SessionRetrieveSerializer
        else:
            return SessionSerializer
```

Reply:
Yes, that's exactly what I was thinking 👍